numpy/core/src/common/simd/neon/arithmetic.h

```diff
@@ -150,4 +159,24 @@
     }
 #endif
 
+// extend sum across vector
```

Comment:
I think "extend then sum" or "sum into a larger type" would be a clearer description here (unless this is common vectorization terminology, in which case could you link a reference as a reply?)

Reply:
> unless this is common vectorization terminology

Yes, it's common. "zero-extend" for unsigned int and "signed-extend" for signed int

Reply:
Right, I understand "extend" and "sum" separately, it's "extend sum" that I find to be confusing - are those words commonly used together like that for vectorization? If not, let's use a longer sentence to make the meaning clear (ie, that it's semantically `sum(extend(vector))` and not `extend(sum(vector))` as the latter would still overflow)

Reply:
> are those words commonly used together like that for vectorization?

I think yes, also the term "wide" too. "extend sum" or "wide sum" both lead to the same meaning.

> let's use a longer sentence to make the meaning clear

I think the following comment can be clear enough:

```C
// wide sum across vector, equivalent to sum(expand(vector))
```

Reply:
@eric-wieser,
I take it back, not common. ARM uses the term `long` instead of `extend`, see https://developer.arm.com/docs/ddi0596/h/simd-and-floating-point-instructions-alphabetic-order/uaddlv-unsigned-sum-long-across-vector

Reply:
I think either `long` or `extend across` is not clear enough, How about "expand the source vector and performs sum reduce, The destination scalar is twice as long as the source vector elements."

Reply:
we will discuss this later during the release of the NPYV doc.